redacted-compiler-plugin/src/main/kotlin/dev/zacsweers/redacted/compiler/RedactedIrVisitor.kt

```diff
@@ -79,16 +75,25 @@ internal class RedactedIrVisitor(
 
       val properties = mutableListOf<Property>()
       val classIsRedacted = declarationParent.hasAnnotation(redactedAnnotation)
+      val supertypeIsRedacted by lazy {
+        declarationParent.getAllSuperclasses().any { it.hasAnnotation(redactedAnnotation) }
```

Comment:
does this include interfaces? I feel like I remember these APIs requiring merging their values

Reply:
Yeah, this includes interfaces and the tests pass at least in the original compiler

Reply:
If you prefer, I could change it to manually check the direct supertypes only without any filtering.

Something like
```kotlin
declarationParent.superTypes
  .map { (it.classifierOrFail.owner as IrClass).hasAnnotation(redactedAnnotation) }
```

Reply:
Nope what you've got works for me 👍